src/js/components/DataChart/Detail.js

```diff
@@ -88,55 +88,70 @@ const Detail = ({
     return res;
   }, [data.length, detailIndex, horizontalProp]);
 
+  const getContent = useCallback(
+    (index) => {
+      if (index !== undefined) {
+        return series
+          .filter(
+            ({ property }) =>
+              ((!activeProperty || activeProperty === property) &&
+                data?.[index]?.[property] !== undefined) ||
+              (axis && axis.x && axis.x.property === property),
+          )
+          .map((serie) => {
+            const axisValue = horizontalProp
+              ? data[index][serie.property]
+              : index;
+            return `${serie.label || serie.property} ${renderValue(
+              serie,
+              axisValue,
+            )}`;
+          })
+          .join(' ');
+      }
+      return undefined;
+    },
+    [activeProperty, axis, data, horizontalProp, renderValue, series],
+  );
+
   useEffect(() => {
     if (detailIndex !== undefined) {
-      const content = series
-        .filter(
-          ({ property }) =>
-            ((!activeProperty || activeProperty === property) &&
-              data?.[detailIndex]?.[property] !== undefined) ||
-            (axis && axis.x && axis.x.property === property),
-        )
-        .map((serie) => {
-          const axisValue = horizontalProp
-            ? data[detailIndex][serie.property]
-            : detailIndex;
-          return `${serie.label || serie.property} ${renderValue(
-            serie,
-            axisValue,
-          )}`;
-        })
-        .join(' ');
-      announce(content);
+      announce(getContent(detailIndex), 'assertive');
     }
-  }, [
-    activeProperty,
-    announce,
-    axis,
-    data,
-    detailIndex,
-    horizontalProp,
-    renderValue,
-    series,
-  ]);
+  }, [announce, detailIndex, getContent]);
```

Comment:
I'm not sure if this effect hook is still needed, when `announce` is called directly from the keyboard handlers.

Reply:
Testing shows it doesn't add anything in Chrome or Safari, so I've removed it.